src/components/ConversationSettings/ConversationAvatarEditor.vue

```diff
@@ -0,0 +1,310 @@
+					:disable-menu="true" />
+				<div v-else class="icon-loading" />
+			</div>
+			<div class="avatar__buttons">
+				<NcButton :aria-label="t('settings', 'Upload profile picture')"
+					@click="activateLocalFilePicker">
+					<template #icon>
+						<Upload :size="20" />
+					</template>
+				</NcButton>
+				<NcButton :aria-label="t('settings', 'Choose profile picture from files')"
+					@click="openFilePicker">
+					<template #icon>
+						<Folder :size="20" />
+					</template>
+				</NcButton>
+				<NcButton v-if="hasAvatar"
+					:aria-label="t('settings', 'Remove profile picture')"
+					@click="removeAvatar">
+					<template #icon>
+						<Delete :size="20" />
+					</template>
+				</NcButton>
+			</div>
+			<span>{{ fileTypeWarning }}</span>
+			<input :id="inputId"
+				ref="input"
+				type="file"
+				:accept="validMimeTypes.join(',')"
+				@change="onChange">
+		</div>
+
+		<!-- Use v-show to ensure early cropper ref availability -->
+		<div v-show="showCropper" class="avatar__container">
+			<VueCropper ref="cropper"
+				class="avatar__cropper"
+				v-bind="cropperOptions" />
+			<div class="avatar__cropper-buttons">
+				<NcButton @click="cancel">
+					{{ t('spreed', 'Cancel') }}
+				</NcButton>
+				<NcButton type="primary"
+					@click="saveAvatar">
+					{{ t('spreed', 'Set as conversation picture') }}
+				</NcButton>
+			</div>
+		</div>
+	</section>
+</template>
+
+<script>
+import VueCropper from 'vue-cropperjs'
+
+import axios from '@nextcloud/axios'
+import { getFilePickerBuilder, showError } from '@nextcloud/dialogs'
+import { generateUrl } from '@nextcloud/router'
+
+// import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+
+import ConversationIcon from '../ConversationIcon.vue'
+
+// eslint-disable-next-line n/no-extraneous-import
+import 'cropperjs/dist/cropper.css'
+
+import Upload from 'vue-material-design-icons/Upload.vue'
+import Folder from 'vue-material-design-icons/Folder.vue'
+import Delete from 'vue-material-design-icons/Delete.vue'
+
+const VALID_MIME_TYPES = ['image/png', 'image/jpeg']
+
+const picker = getFilePickerBuilder(t('spreed', 'Choose your profile picture'))
+	.setMultiSelect(false)
+	.setMimeTypeFilter(VALID_MIME_TYPES)
+	.setModal(true)
+	.setType(1)
+	.allowDirectories(false)
+	.build()
+
+export default {
+	name: 'ConversationAvatarEditor',
+
+	components: {
+		Delete,
+		Folder,
+		// NcAvatar,
+		NcButton,
+		Upload,
+		VueCropper,
+		ConversationIcon,
+	},
+
+	props: {
+		conversation: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			// Todo: get real value here
+			showCropper: false,
+			loading: false,
+			validMimeTypes: VALID_MIME_TYPES,
+			cropperOptions: {
+				aspectRatio: 1 / 1,
+				viewMode: 1,
+				guides: false,
+				center: false,
+				highlight: false,
+				autoCropArea: 1,
+				minContainerWidth: 300,
+				minContainerHeight: 300,
+			},
+		}
+	},
```

Comment:
`validMimeTypes` and `cropperOptions` are not a part of the component instance's reactive state (data). They never change and are actually constants. 

I know that it is a very common practice to add constants to `data` to add them into component instance properties, but in my opinion, it is not good practice because:
- It makes them feel like a changeable state, not a constant
- Vue (especially Vue 2) traverses the object, redefines all the properties, and creates an observer for watching, `Dep`s instances for each property, making them reactive. It's a bit worse in performance.

```html
<template>
  <!-- In template we can take options from $options -->
  <VueCropper v-bind="$options.CROPPER_OPTIONS" />
</template>

<script>
const VALID_MIME_TYPES = ['image/png', 'image/jpeg']
const CROPPER_OPTIONS = { /* ... */ }

export default {
	name: 'ConversationAvatarEditor',
	
	VALID_MIME_TYPES,
	CROPPER_OPTIONS,
	
	methods: {
	  foo(type) {
	    // In script we can use constants as it is
	    VALID_MIME_TYPES.includes(type)
	  }
	}
}
</script>
```

But as it is a common practice in Options API, I'm fine with any option.